src/sparseml/pytorch/utils/sparsification_info/configs.py

```diff
@@ -12,20 +12,168 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from pydantic import BaseModel
+from collections import Counter, defaultdict
+from typing import Any, Dict, Tuple
+
+import torch.nn
+from pydantic import BaseModel, Field
+
+from sparseml.pytorch.utils.sparsification_info.helpers import (
+    get_dtype,
+    get_leaf_operations,
+    is_quantized,
+)
+
+
+__all__ = [
+    "SparsificationSummaries",
+    "SparsificationPruning",
+    "SparsificationQuantization",
+]
+
+
+class CountAndPercent(BaseModel):
+    count: int = Field(description="The count of items")
+    percent: float = Field(description="The percentage of those items out of the total")
 
 
 class SparsificationSummaries(BaseModel):
-    pass
+    """
+    A model that contains the sparsification summaries for a torch module.
+    """
+
+    quantization: CountAndPercent = Field(
+        description="A model that contains the number of "
+        "operations/the percent of operations that are quantized."
+    )
+    pruning: CountAndPercent = Field(
+        description="A tuple that displays of the number of "
+        "parameters/the percent of parameters that are pruned."
+    )
+    count_parameters: Dict[str, int] = Field(
+        description="A dictionary that maps the name of a parameter "
+        "to the number of elements in that parameter."
+    )
+    count_operations: Dict[str, int] = Field(
+        description="A dictionary that maps the name of an operation "
+        "to the number of times that operation is used in the model."
+    )
+
+    @classmethod
+    def from_module(
+        cls,
+        module=torch.nn.Module,
+        pruning_thresholds: Tuple[float, float] = (0.05, 0.99),
```

Comment:
need higher than 99%
```suggestion
        pruning_thresholds: Tuple[float, float] = (0.05, 1 - 1e-9),
```